units/patch_intent.go

```diff
@@ -367,18 +362,6 @@ func (j *patchIntentProcessor) finishPatch(ctx context.Context, patchDoc *patch.
 		return errors.New("patch has no build variants or tasks")
 	}
 
-	if shouldTaskSync := len(patchDoc.SyncAtEndOpts.BuildVariants) != 0 || len(patchDoc.SyncAtEndOpts.Tasks) != 0; shouldTaskSync {
-		patchDoc.SyncAtEndOpts.VariantsTasks = patchDoc.ResolveSyncVariantTasks(patchedProject.GetAllVariantTasks())
-		// If the user requested task sync in their patch, it should match at least
-		// one valid task in a build variant.
-		if len(patchDoc.SyncAtEndOpts.VariantsTasks) == 0 {
-			j.gitHubError = NoSyncTasksOrVariants
```

Comment:
we can remove NoSyncTasksOrVariants as well

Reply:
Ty! I still find it ridiculous that the linter official doesn't consider const in blocks like this unused. Removed